### PR TITLE
Adds Support for BigInts in GraphQL SDL, Schema, Type Generation, and Tests + Scenarios (Scaffolding)

### PR DESCRIPTION
**Pull request by dthyresson**
Fix https://github.com/redwoodjs/redwood/issues/3152#issue-957000543
No need for workaround for BigInt scalar as seen in https://github.com/redwoodjs/redwood/pull/3884

Support BigInt scalars in GraphQL SDL, Prisma, and scaffolding (tests, scenarios)

The issue lies with Prisma and that one needs to tell it how to serialize a BigInt.

See: https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields#serializing-bigint

This PR:

* Adds the BigInt scalar and resolver to the root schema and type generation
* Adds random bigints to test data
* Handler deserializing the bigint in scenario generation to pass to Prisma in its create/update arguments

image

image

image

image

### CODE REVIEW

**Comment by dthyresson** [comment]
Also, this is nice to have very large autoincrement ids in Postgres using big serial:

image

**Comment by dthyresson** [comment]
I am taking back this issue into draft as having done some exploration into graphql-scalars, I found their BigInt implementation:

https://github.com/Urigo/graphql-scalars/blob/78f51ed9fd5c7c4d4787cf5dd0779a5b5d97adb6/src/scalars/BigInt.ts#L16

which has some code to do serialization of BigInt a little differently and I may want to use their approach.

**Comment by dthyresson** [comment]
> I am taking back this issue into draft

Did some closer review and am fine with my implementation. The scalar example for de/serialization is a bit different and could not leverage it and be isolated for just the scenario output.

Ready for review.

File: packages/cli/src/commands/generate/service/__tests__/scenario.test.js [hunk 1]
```diff
@@ -38,6 +38,39 @@ describe('the scenario generator', () => {
     ])
   })
 
+  test('parseSchema returns an object with BigINt scalar fields', async () => {
```
Comment (cannikin) [inline packages/cli/src/commands/generate/service/__tests__/scenario.test.js:41]: *tiny* typo!  `BigINt`

**Review by cannikin — APPROVED** [review]

**Comment by cannikin** [comment]
Okay so summing up, `BigInt`s will be `String`s within Scenarios, but everywhere else they stay native `BigInt`s now?

> The issue lies with Prisma and that one needs to tell it how to serialize a BigInt.

Are you saying here that's what the issue was with scenarios, and now it's fixed? Or a note to us (and for the docs) that people need to make sure and use that custom `Json.stringify()` function if they have a database table with a `BigInt`?

**Comment by dthyresson** [comment]
> Are you saying here that's what the issue was with scenarios, and now it's fixed?

Yes, I have fixed the issue and now using BigInts in the Prisma schema will generate valid scenario data.

**Pull request merged**